examples/slack-clone/nextjs-slack-clone/full-schema.sql

```diff
@@ -141,24 +140,90 @@ alter publication supabase_realtime add table public.channels;
 alter publication supabase_realtime add table public.messages;
 alter publication supabase_realtime add table public.users;
 
--- DUMMY DATA
-insert into public.users (id, username)
-values
-    ('8d0fd2b3-9ca7-4d9e-a95f-9e13dded323e', 'supabot');
+/**
+ * AUTH HOOKS
+ * Create an auth hook to add a custom claim to the access token jwt.
+ */
+
+-- Enable the "plv8" extension
+-- https://supabase.com/docs/guides/database/extensions/plv8?database-method=sql#enable-the-extension
+create extension plv8;
+
+-- Create the auth hook function
+-- https://supabase.com/docs/guides/auth/auth-hooks?language=add-metadata-claim-plv8#hook-custom-access-token
+create or replace function custom_access_token_hook(event jsonb)
+returns jsonb
+language plv8
```

Comment:
this will be a lot faster if it's pure `sql`! it also looks like it could be marked as `immutable`?
<img width="1379" alt="image" src="https://github.com/supabase/supabase/assets/10214025/21549dd6-e1a0-4587-b9fc-92ec979f639a">

Reply:
Okay, I've changed it to use plpgsql. Would you mind reviewing the function?